services/datamanager/builtin/builtin.go

```diff
@@ -579,9 +438,36 @@ func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
 		}
 	}
 
+	svc.syncDisabled = svcConfig.ScheduledSyncDisabled
+	svc.syncIntervalMins = svcConfig.SyncIntervalMins
+	svc.additionalSyncPaths = svcConfig.AdditionalSyncPaths
+	// TODO: don't cancel in progress arbitrary file uploads on Update unless sync has been disabled or
```

Comment:
Update() will get called for _any_ robot config change, regardless of whether it's anything within data capture/sync that's changed. 

This seems to cancel and restart all sync routines if there's any config change in Update, and similarly re-initializes a syncer on any config change. 

This is why previously only changed the syncer if it was toggled in the update, or the sync interval mins changed. 

Thinking about the interaction between manual Sync and the scheduled sync interval: if the user changes the sync interval while doing a manual sync, we'll cancel sync processes & init the new syncer, but not vice versa. I think this is fine if documented somewhere. Don't want to add any more complications between the interaction of the two functionalities.

Reply:
Ah yeah sorry totally forgot to go back and do this TODO. accomplished this by:
- Cancelling the sync background routine. This doesn't stop the syncer, kill anything in progress, or rebuild a syncer. It just stops the cron that calls .SyncDirectory.
- If sync is enabled:
   - If syncer is nil, make one
   - In any case: start off a new sync background routine
- else if sync is disabled, close the syncer
   
This won't cause leaked goroutines, because even after the background routine cancel fn is called, we wait on it to finish on dmsvc.Close because it uses the backgroundWorkers waitgroup. It doesn't cause in progress syncs to cancel, because we only close the syncer (killing in progress uploads) if sync is now disabled.

This does leave the problem that an update will kill manual sync if sync is disabled in the config. But I think there's some discussion for how we want that to behave. If I have a robot, and I hit manual sync, do I have anyway to change my mind or stop it? Currently, you can disable sync. But if we change that, there won't be any cancellation options for manual sync beyond turning off the robot.

Reply:
Again more of a product question, but after thinking about it I think it might be reasonable to allow people to have sync enabled but no interval configured (interval == 0). 

Then we won't do scheduled syncs, but the user can do manual syncs. They can stop them by turning of syncing.
If sync is not enabled, we can disallow manual syncs.

Initial thoughts on if that seem like a reasonable UX? Not a this PR thing and probably warrants a bigger talk

Reply:
I agree that in the high-level product sense, it makes complete sense to require the syncer to be enabled (but scheduled sync set to 0) in order to use manual sync. Even if we'd run this from the terminal rather than the app in the future, it's intuitive that the config's sync service should be enabled in order to use _either_ manual or scheduled sync.

Could you add a comment above CancelSyncBackgroundRoutine that it doesn't kill any in-progress syncing, only the crawling for newly added files? Or if there's a better method name to reflect that.

> In any case: start off a new sync background routine

If so, I believe the side effect is that if my ticker is on a 10min schedule and I update my config for something unrelated to data, I'll restart the ticker. Slightly unexpected but totally fine for simplicity sake - I would add a comment explaining that Update will reset the timer of sync and perhaps a TODO/ticket to make it consistent? Wouldn't add to this PR.

Reply:
> Could you add a comment above CancelSyncBackgroundRoutine that it doesn't kill any in-progress syncing, only the crawling for newly added files? Or if there's a better method name to reflect that.

Renamed and added comments above both functions. Do you think that name is more clear?

> If so, I believe the side effect is that if my ticker is on a 10min schedule and I update my config for something unrelated to data, I'll restart the ticker. Slightly unexpected but totally fine for simplicity sake - I would add a comment explaining that Update will reset the timer of sync and perhaps a TODO/ticket to make it consistent? Wouldn't add to this PR.

Great call out, definitely not desired behavior. I made [this](https://viam.atlassian.net/browse/DATA-861) ticket to address it and added a TODO